internal/systemd/mock_test.go

```diff
@@ -0,0 +1,203 @@
+			[]interface{}{"Unit not-a-service.service not found."},
+		)
+	}
+
+	return s.emitJobSignals(name), nil
+}
+
+func (s *systemdBus) StopUnit(name string, _ string) (dbus.ObjectPath, *dbus.Error) {
+	if name == "not-a-service.service" {
+		return dbus.ObjectPath("/"), dbus.NewError(
+			fmt.Sprintf("%s.NoSuchUnit", consts.SystemdDbusRegisteredName),
+			[]interface{}{"Unit not-a-service.service not found."},
+		)
+	}
+
+	return s.emitJobSignals(name), nil
+}
+
+func (s *systemdBus) EnableUnitFiles(names []string, _ bool, _ bool) (bool, [][]string, *dbus.Error) {
+	if len(names) != 1 {
+		panic("method is only expected to be called with a single name")
+	}
+
+	if name := names[0]; name == "not-a-service.service" {
+		return false, nil, dbus.NewError(
+			fmt.Sprintf("%s.NoSuchUnit", consts.SystemdDbusRegisteredName),
+			[]interface{}{"Unit not-a-service.service not found."},
+		)
+	}
+
+	return true, [][]string{{"symlink", "/from/path", "/to/path"}}, nil
+}
+
+func (s *systemdBus) DisableUnitFiles(names []string, _ bool) ([][]string, *dbus.Error) {
+	if len(names) != 1 {
+		panic("method is only expected to be called with a single name")
+	}
+
+	if name := names[0]; name == "not-a-service.service" {
+		return nil, dbus.NewError(
+			fmt.Sprintf("%s.NoSuchUnit", consts.SystemdDbusRegisteredName),
+			[]interface{}{"Unit not-a-service.service not found."},
+		)
+	}
+
+	return [][]string{{"symlink", "/from/path", "/to/path"}}, nil
+}
+
+func (s *systemdBus) Reload() *dbus.Error {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	if s.reloadError {
+		return dbus.MakeFailedError(fmt.Errorf("reload error"))
+	}
+
+	return nil
+}
+
+func (s *systemdBus) setErrorOnReload() {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	s.reloadError = true
+}
+
+func (s *systemdBus) emitJobSignals(name string) dbus.ObjectPath {
+	s.mu.Lock()
+	defer s.mu.Unlock()
+
+	// Increment job ID
+	s.nextJobID++
+	jobPath := dbus.ObjectPath(fmt.Sprintf("%s/Job/%d", consts.SystemdDbusObjectPath, s.nextJobID))
+
+	// Emit JobNew signal
+	err := s.conn.Emit(
+		dbus.ObjectPath(consts.SystemdDbusObjectPath),
+		fmt.Sprintf("%s.JobNew", consts.SystemdDbusManagerInterface),
+		s.nextJobID, jobPath, name,
+	)
+	if err != nil {
+		log.Fatalf("failed to emit JobNew signal: %v", err)
+	}
+
+	jobStatus := "done"
+	if name == "fail-to-start-stop.service" {
+		jobStatus = "failed"
+	}
+
+	// Emit JobRemoved signal
+	err = s.conn.Emit(
+		dbus.ObjectPath(consts.SystemdDbusObjectPath),
+		fmt.Sprintf("%s.JobRemoved", consts.SystemdDbusManagerInterface),
+		s.nextJobID, jobPath, name, jobStatus,
+	)
+	if err != nil {
+		log.Fatalf("failed to emit JobRemoved signal: %v", err)
+	}
+
+	return jobPath
+}
+
+func TestMain(m *testing.M) {
+	// export systemd structure
+	defer testutils.StartLocalSystemBus()()
+
+	debug := flag.Bool("verbose", false, "Print debug log level information within the test")
+	flag.Parse()
+
+	var connOpts []dbus.ConnOption
+	if *debug {
```

Comment:
We have a card for this (we should handle debug logs consistently), but let’s get that in.